agent/agentssh/agentssh.go

```diff
@@ -428,6 +435,19 @@ func (s *Server) startPTYSession(session ptySession, magicTypeLabel string, cmd
 	return nil
 }
 
+func (s *Server) showServiceBanner(ctx context.Context, session io.Writer) error {
+	banner, err := s.GetServiceBanner(ctx)
+	if err != nil {
+		return err
+	}
+	if banner.Enabled && banner.Message != "" {
+		// The banner supports Markdown so we might want to parse it but Markdown is
+		// still fairly readable in its raw form.
+		_, err = io.WriteString(session, banner.Message+"\n\n\r")
```

Comment:
Consider handling the error immediately to avoid a chance of code sneaking in between the if branch and `return err`.

Reply:
Oh wow yeah that is way smarter.